NEWS.md

```diff
@@ -1,59 +1,58 @@
-   to automatically deduce the timezone value on the assumption that the message
-   is received in near real time. (#2818)
- * Send Server Name Identification (SNI) information with `transport(tls)`.
-   Enable it by setting the `sni(yes)` option in the `tls` block in your
-   `destination`. (#2930)
+ * `http-destination`: Users now can specify the action for any HTTP result code.
+   Use with `response-action(response_code => action)` in your http block.
+   Available actions are: `success`, `retry`, `drop` and `disconnect`. (#3007)
+ * `syslog-ng-cfg-db`: Added a new script, which can provide the options of
+   sources and destinations queried by the user. This tool can make the configuration
+   of syslog-ng a lot easier. Use with `./syslog-ng-cfg-db.py` from the
+   `contrib/config_database` dir.(#2997)
+ * `redis-destination`: Improved the performance by 2 orders of magnitude.
+   In our labor environment, now it operates at 25k EPS. (#2972)
 
 ## Features
 
- * `templates`: change the `$LOGHOST` macro to honour `use-fqdn()` (#2894)
- * Define `syslog-ng-sysconfdir` (#2932)
- * `dqtool`: add assign dqfile to persist file feature (#2872)
+ * `create-dirs()`: Added to `pipe()` source/destination, and standardize the behavior.
+   (#3018, #2635)
+ * `default-network-drivers`: Added `max-connections()` option, to change the limit
+    from 10. (#2961)
+ * `checkpoint`: Added support for timezone value at the end of timestamps. (#3033)
+ * `filter/rewrite`: Added `disable-jit` flag to disable JIT PCRE compilation. (#2992, #2986)
+ * `syslog-ng-ctl`: Add `export-config-graph` option to visualize config graph. (#2990)
+ * `build/dbld`: Readded CentOS 6 support. (#2860, #2971, #3028)
+ * `python`: Added Python 3.8 support. (#3017)
 
 ## Bugfixes
 
- * Fix backtick subsitution of defines/environment variables in the main configuration file. (#2906, #2909)
- * Fix SCL block parameter substitution of quoted escaped newline (#2901)
- * `python, diskq, random-generator source`: crash after failed reload (#2907)
- * Fix crash at shutdown on 32bit systems (#2893, #2895)
- * Invalidate the value of the `LEGACY_MSGHDR` macro in case either the `PID` or the `PROGRAM`
-   macros are `unset()` using a `rewrite` rule. Previously `LEGACY_MSGHDR` would retain the old values. (#2896)
- * on 32bit platform `diskq` ftruncate could fail due to size 32/64 interface (#2892)
- * Support new tzdata format, starting from version 2009.XXX, in tzinfo parser. (#2898)
- * `udp, udp6, tcp, tcp6, syslog, network destination`: Correctly detect and set `IP_MULTICAST_TTL`
-   in case of multicast ip address (#2905)
- * Fix hostname resolve on systems with only the loopback network interface configured (#2933)
- * `wildcard-file()`: Add `multi-line()`, `pad_size()` and `multi-line-mode()` option validation. (#2922)
- * `kafka-c`: Fix multiple memleaks (#2944)
+ * `tls`: Fixed an infinite loop which occured, when a `TLS` connection broke. (#3026, #3009)
```

Comment:
there was a bugfix around stomp (a buffer over read issue)

Reply:
Sorry about the PR name, it was a really bad idea:

Light property based test fixes. #2988


This contains multiple fixes, the interesting ones:
```
stomp: fix buffer over-read

stomp() operates on input that is read from a socket.
It is not safe to assume anything about the input.
```

maybe this one:
```
pseudofile: fix null pointer dereference

The template() option of pseudofile() is mandatory.
pseudofile() is used on pseudo file systems, such as /sys, /proc, and
/dev, so it requires a special output format that has to be specified
manually by the user.
```

Reply:
```
 * `stomp`: Fixed a buffer over-read on connection. (#2988)
 * `pseudofile`: Fixed a crash, when `template()` option is not set. (#2988)
 * `wildcard-source`: Fixed a crash, when `max-files()` was set to 0. (#2988)
```

Reply:
Thanks!